src/recordlinker/utils/normalize.py

```diff
@@ -0,0 +1,14 @@
+import re
+import unicodedata
+
+
+def normalize_text(text: str) -> str:
+    """
+    Normalize text for comparison by removing non-alphanumeric characters, converting
+    to lowercase, and removing all whitespace (trailing, leading, and internal).
+    """
+    text = "".join(
+        c for c in unicodedata.normalize("NFKD", text) if unicodedata.category(c) != "Mn"
```

Comment:
nitpick; I honestly don't know which is better, but I think converting to ascii and back would work as well.  Just adding this as something to think about, on the surface I don't see anything wrong with this implementation.

```python
text = unicodedata.normalize("NFKD", text).encode('ASCII', 'ignore').decode('ASCII')
```

Reply:
Interestingly, the ASCII approach appears to be generally a slower, particularly for the address example that does not contain any accents.

![image](https://github.com/user-attachments/assets/f5c9a682-775e-41b0-ad3c-f61258062228)

Reply:
In either case, it's negligible, so really I'm not too concerned.  But I think the ASCII version is a power of 10 smaller than the original for each test cases (Note the differences in `e-05` vs `e-06`)

Reply:
oof that's what multi-tasking will get me 😓